src/doc/reference.md

```diff
@@ -2070,6 +2070,9 @@ macro scope.
    trait of the same name. `{Self}` will be replaced with the type that is supposed
    to implement the trait but doesn't. To use this, the `on_unimplemented` feature gate
    must be enabled.
+- `must_use` - on structs and enums, requires that instances are either consumed
+   or passed outside the crate. Using the full version `#[must_use = "message"]`
```

Comment:
@Manishearth On second thought, you're absolutely right. I guess I was reading too into the lint that emitted the warning.

@steveklabnik I like that! I'll get that written in.